compatibility/bazel_tools/daml_ledger/Main.hs

```diff
@@ -119,33 +123,45 @@ main = do
       , fetchTest sdkVersion getTools
       ]
 
--- | Test `daml ledger upload-dar --access-token-file`
+-- | Test `daml ledger list-parties --access-token-file`
 authenticatedUploadTest :: SdkVersion -> IO Tools -> TestTree
 authenticatedUploadTest sdkVersion getTools = do
-  let sharedSecret = "TheSharedSecret"
-  let getSandboxConfig = do
+  withSandbox getSandboxConfig $ \getSandboxPort ->  testGroup "authentication" $
+    [ testCase "Bearer prefix" $ do
+          Tools{..} <- getTools
+          port <- getSandboxPort
+          withTempDir $ \deployDir -> do
+            withCurrentDirectory deployDir $ do
+              let tokenFile = deployDir </> "secretToken.jwt"
+              -- The trailing newline is not required but we want to test that it is supported.
+              writeFileUTF8 tokenFile ("Bearer " <> makeSignedJwt sharedSecret <> "\n")
+              callProcessSilent daml
+                [ "ledger", "list-parties"
+                , "--access-token-file", tokenFile
+                , "--host", "localhost", "--port", show port
+                ]
+    ] <>
+    [ testCase "no Bearer prefix" $ do
+          Tools{..} <- getTools
+          port <- getSandboxPort
+          withTempDir $ \deployDir -> do
+            withCurrentDirectory deployDir $ do
+              let tokenFile = deployDir </> "secretToken.jwt"
+              -- The trailing newline is not required but we want to test that it is supported.
+              writeFileUTF8 tokenFile (makeSignedJwt sharedSecret <> "\n")
+              callProcessSilent daml
+                [ "ledger", "list-parties"
+                , "--access-token-file", tokenFile
+                , "--host", "localhost", "--port", show port
+                ]
+    | sdkVersion == SdkVersion "0.0.0"
```

Comment:
Would that make sense?
```suggestion
    | sdkVersion == SdkVersion "0.0.0" || sdkVersion >= SdkVersion "1.1.0"
```
Maybe for a different version than `1.1.0`?

Reply:
Doesn’t quite work since the ordering is wrong. This really needs SemVer parsing to be correct.